src/Cms/FileBlueprint.php

```diff
@@ -116,6 +118,79 @@ public function acceptMime(): string
 		return '*';
 	}
 
+
+	/**
+	 * Returns the list of all accepted file extensions
+	 * for file upload or `*` if all extensions are allowed
+	 *
+	 * If a mime type is specified in the blueprint, the extensions and types options are ignored for the browser.
+	 * Extensions and Types, however, are still used to validate an uploaded file on the server.
+	 * This behavior might change in the future to better represent what file extensions are actually allowed.
+	 *
+	 * If no mime type is specified, the intersection between manually defined extensions and the Kirby "file types" is returned.
+	 * If the intersection is empty, an empty string is returned.
+	 * This behavior might change in the future to instead return the union of mime, extensions and types.
+	 *
+	 * @since 4.2.0
+	 * @return string
+	 */
+	public function acceptAttribute(): string
+	{
+		// don't disclose the specific default types
+		if ($this->defaultTypes === true) {
+			return '*';
+		}
+
+		$accept = $this->accept();
+
+		// get extensions from "mime" option
+		if (is_array($accept['mime']) === true) {
+			// determine the extensions for each MIME type
+			$extensions = array_map(
+				fn ($pattern) => Mime::toExtensions($pattern, true),
+				$accept['mime']
+			);
+
+			$fromMime = array_unique(array_merge(...array_values($extensions)));
+
+			// return early to ignore the other options
+			return implode(',', array_map(fn ($ext) => ".$ext", $fromMime));
+		}
+
+		$restrictions = [];
+
+		// get extensions from "type" option
+		if (is_array($accept['type']) === true) {
+			$extensions = array_map(
+				F::typeToExtensions(...),
+				$accept['type']
+			);
+
+			// F::typeToExtensions might return null instead of empty arrays,
+			// we need to filter those out
+			$fromType = array_merge(...array_values(array_filter($extensions)));
+			$restrictions[] = $fromType;
+		}
+
+		// get extensions from "extension" option
+		if (is_array($accept['extension']) === true) {
+			$fromExtension = $accept['extension'];
+			$restrictions[] = $fromExtension;
+		}
+
+		// intersect all restrictions
+		if (count($restrictions) > 1) {
+			$list = array_intersect(...$restrictions);
+		} else {
+			$list = $restrictions[0];
+		}
```

Comment:
Wouldn't this fail for an empty restrictions array? We can also avoid the if/else statement by using `match`

```php
$list = match (count($restroctions)) {
    0 => [],
    1 => $restrictions[0],
    default => array_intersect(...$restrictions)
};
```